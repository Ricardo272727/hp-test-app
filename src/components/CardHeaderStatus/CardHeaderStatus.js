import React from "react";
import "./CardHeaderStatus.scss";

export const CardHeaderStatus = ({
  alive = false,
  hogwartsStudent = false,
}) => (
  <span className="card-header-status">
    {alive ? "VIVO" : "FINADO"} / {hogwartsStudent ? "ESTUDIANTE" : "STAFF"}
  </span>
);
