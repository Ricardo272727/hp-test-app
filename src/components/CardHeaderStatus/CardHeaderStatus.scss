@import "../../scss/colors.scss";

.card-header-status {
  color: $gray;
  font-size: 14px;
}
