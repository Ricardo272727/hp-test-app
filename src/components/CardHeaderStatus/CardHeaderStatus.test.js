/**
 * @jest-environment jsdom
 */
import React from "react";
import {
  cleanup,
  fireEvent,
  getByTestId,
  render,
  screen,
} from "@testing-library/react";
import { CardHeaderStatus } from "./CardHeaderStatus";
import renderer from 'react-test-renderer';

test("check card header status display FINADO / ESTUDIANTE", () => {
  const fn = jest.fn((e) => {});
  let component = renderer.create(<CardHeaderStatus alive={false} hogwartsStudent={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("check card header status display VIVO / STAFF ", () => {
  const fn = jest.fn((e) => {});
  let component = renderer.create(<CardHeaderStatus alive={true} hogwartsStudent={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


