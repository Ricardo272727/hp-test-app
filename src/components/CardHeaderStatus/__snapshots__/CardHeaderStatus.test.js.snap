// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check card header status display FINADO / ESTUDIANTE 1`] = `
<span
  className="card-header-status"
  data-testid="card-header-status"
>
  FINADO
   / 
  ESTUDIANTE
</span>
`;

exports[`check card header status display VIVO / STAFF  1`] = `
<span
  className="card-header-status"
  data-testid="card-header-status"
>
  VIVO
   / 
  STAFF
</span>
`;
